### PR TITLE
OCPBUGS-15256: Sync openshift/api to reduce installerpod configmap based file permissions to 0600

### PR DESCRIPTION
**Pull request by ingvagabund**
Pulling https://github.com/openshift/library-go/pull/1546

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -42,6 +42,8 @@ require (
 	k8s.io/utils v0.0.0-20230406110748-d93618cff8a2
 )
 
+replace github.com/openshift/library-go => github.com/ingvagabund/library-go v0.0.0-20230621135628-6de88d61f435
```
Comment (tjungblu) [inline go.mod:45]: are you sure about that replacement? ;)

Comment (tjungblu) [reply go.mod:45]: ah you're still testing, apologies!

**Comment by ingvagabund** [comment]
From the installer etcd pod:
```
    - --configmaps=etcd-pod
    - --configmaps=etcd-serving-ca
    - --configmaps=etcd-peer-client-ca
    - --configmaps=etcd-metrics-proxy-serving-ca
    - --configmaps=etcd-metrics-proxy-client-ca
    - --configmaps=etcd-endpoints
```

Permissions of all files produced from ^^ config maps:
```
sh-4.4$ for i in $(oc get pods -n openshift-etcd -l app=etcd -o name); do oc exec -n openshift-etcd $i -- stat -c %a /etc/kubernetes/static-pod-resources/configmaps/etcd-pod/pod.yaml; done
600
600
600
sh-4.4$ for i in $(oc get pods -n openshift-etcd -l app=etcd -o name); do oc exec -n openshift-etcd $i -- stat -c %a /etc/kubernetes/static-pod-resources/configmaps/etcd-pod/version; done
600
600
600
sh-4.4$ for i in $(oc get pods -n openshift-etcd -l app=etcd -o name); do oc exec -n openshift-etcd $i -- stat -c %a /etc/kubernetes/static-pod-resources/configmaps/etcd-pod/forceRedeploymentReason; done
600
600
600
sh-4.4$ for i in $(oc get pods -n openshift-etcd -l app=etcd -o name); do oc exec -n openshift-etcd $i --  stat -c %a /etc/kubernetes/static-pod-resources/configmaps/etcd-serving-ca/ca-bundle.crt; done
600
600
600
sh-4.4$ for i in $(oc get pods -n openshift-etcd -l app=etcd -o name); do oc exec -n openshift-etcd $i --  stat -c %a /etc/kubernetes/static-pod-resources/configmaps/etcd-peer-client-ca/ca-bundle.crt; done
600
600
600
sh-4.4$ for i in $(oc get pods -n openshift-etcd -l app=etcd -o name); do oc exec -n openshift-etcd $i --  stat -c %a /etc/kubernetes/static-pod-resources/configmaps/etcd-metrics-proxy-serving-ca/ca-bundle.crt; done
600
600
600
sh-4.4$ for i in $(oc get pods -n openshift-etcd -l app=etcd -o name); do oc exec -n openshift-etcd $i --  stat -c %a /etc/kubernetes/static-pod-resources/configmaps/etcd-metrics-proxy-client-ca/ca-bundle.crt; done
600
600
600
sh-4.4$ for i in $(oc get pods -n openshift-etcd -l app=etcd -o name); do oc exec -n openshift-etcd $i --  stat -c %a /etc/kubernetes/static-pod-resources/configmaps/etcd-endpoints/48214bae99c3337d; done
600
600
600
sh-4.4$ for i in $(oc get pods -n openshift-etcd -l app=etcd -o name); do oc exec -n openshift-etcd $i --  stat -c %a /etc/kubernetes/static-pod-resources/configmaps/etcd-endpoints/ad05a8de0369da57; done
600
600
600
sh-4.4$ for i in $(oc get pods -n openshift-etcd -l app=etcd -o name); do oc exec -n openshift-etcd $i --  stat -c %a /etc/kubernetes/static-pod-resources/configmaps/etcd-endpoints/ff07015b5bcc3271; done
600
600
600
```

**Comment by tjungblu** [comment]
our newly added recovery test in https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-etcd-operator/1065/pull-ci-openshift-cluster-etcd-operator-master-e2e-aws-etcd-recovery/1673328405812088832 seems to be :green_circle: 

so I think this should work fine :)

**Comment by ingvagabund** [comment]
@tjungblu awesome :) In that case would you please lgtm https://github.com/openshift/library-go/pull/1546?

**Comment by tjungblu** [comment]
/lgtm

thanks :)

**Comment by tjungblu** [comment]
overriding the SNO job, due to unrelated networking failures

/override ci/prow/e2e-aws-ovn-single-node

**Pull request merged**